coops-ui/tests/unit/Directors.spec.ts

```diff
@@ -168,110 +137,89 @@ describe('Directors.vue', () => {
     expect(directorListUI[1].innerHTML).toContain('<span>Cease</span>')
   })
 
-  it('buttons/actions are disabled when the AGM date has not been set', () => {
-    // clear AGM Date in AGMDate component
-    childvm.$store.state.agmDate = null
+  it('disables buttons/actions when AGM Date is invalid', done => {
+    // invalidate AGM Date
+    vm.$store.state.agmDateValid = false
 
     Vue.nextTick(() => {
-      var directorListUI = childvm.$el.querySelectorAll('.director-list .container')
-
       // confirm that flag is set correctly
-      expect(childvm.agmEntered).toEqual(false)
+      expect(vm.agmDateValid).toEqual(false)
+
+      const directorListUI = vm.$el.querySelectorAll('.director-list .container')
 
       // check that buttons are disabled (checks first button in first director, plus the Add New Director button)
-      expect(
-        directorListUI[0].querySelector('.cease-btn').attributes.getNamedItem('disabled').value
-      ).toEqual('disabled')
-      expect(
-        childvm.$el.querySelector('.new-director-btn').attributes.getNamedItem('disabled').value
-      ).toEqual('disabled')
+      expect(directorListUI[0].querySelector('.cease-btn').disabled).toBe(true)
+      expect(vm.$el.querySelector('.new-director-btn').disabled).toBe(true)
+
+      done()
```

Comment:
I see that done() was removed from afterEach() but added to the end of most of these tests. Do you know if that is necessary? If we can keep it in afterEach() that would be better for code re-use, rather than repetition.

Reply:
It doesn't work in afterEach() because it's not called in the context of the it() test itself.
As you can see in the new code, 'done' is a callback function to tell the it() when it's complete (which is why it's called the bottom of the test).
But if you call done() in afterEach(), you are only telling afterEach() when it's done.